SECURITY.md

```diff
@@ -2,58 +2,41 @@
 
 ## Supported Versions
 
-Vyper is currently in limited beta.
-This means that we only support the latest release and that you may encounter issues using it.
-It is un-audited software, use with caution.
+- each Vyper version in the range `<v0.1.0-beta.1, 0.3.10>` contains a high severity vulnerability
+  - if developing with such versions, please read https://github.com/vyperlang/vyper/security/advisories to learn how to work around the vulnerabilities
+- as of May 2024, the `0.4.0` release is the most secure and the most comprehensively reviewed one and is recommended for use in production environments
+- if a compiler vulnerability is found, a new compiler version with a patch will be released. The vulnerable version itself is not updated.
 
-## Audit reports
+## Compiler Audits
 
-Vyper is constantly changing and improving.
-This means the latest version available may not be audited.
-We try to ensure the highest security code possible, but occasionally things slip through.
+- Vyper is conducts recurring security audits with multiple firms. Additionally, a competitive audit with [CodeHawks](https://www.codehawks.com/contests/cll5rujmw0001js08menkj7hc) was conducted during the fall of 2023.
```

Comment:
We can list the names IMO.

Reply:
i'd like to minimize the need to update the document in the future months/years, as such, I want to hide the implementation behind the interface (the link 😉)